packages/waas/src/intents/base.ts

```diff
@@ -11,9 +11,21 @@ export type SignedIntent<T> = Omit<RawIntent, 'data'> & { data: T }
 const INTENTS_VERSION = 1
 const VERSION = `${INTENTS_VERSION} (Web ${PACKAGE_VERSION})`
 
+// The result of localTime - serverTime, in seconds
+let timeDrift: number | undefined
+
+export function getTimeDrift() {
+  return timeDrift
```

Comment:
can you add check for window.sessionStorage (if exists), and try to get from there if it does, otherwise use just the in-memory. sessionStorage is preserved between page reloads, it will just save some unnecessary requests and its a few extra lines to do here and `updateTimeDrift`

Reply:
like so..

```ts
function isSessionStorageAvailable() {
  return typeof window === 'object' && typeof window.sessionStorage === 'object'
}

const timeDriftKey = '@sequence.timeDrift'

export function getTimeDrift() {
  if (isSessionStorageAvailable()) {
    return window.sessionStorage.get('timeDriftKey') // TODO: convert to number
  } else {
    return timeDrift
  }
}


// etc..
```